spark/src/main/scala/org/apache/spark/sql/delta/skipping/MultiDimClustering.scala

```diff
@@ -90,7 +91,15 @@ trait SpaceFillingCurveClustering extends MultiDimClustering {
         .repartitionByRange(approxNumPartitions, col(repartitionKeyColName))
     }
 
-    repartitionedDf.drop(repartitionKeyColName)
+    val optionallySortedDf = if (sort) {
+      repartitionedDf.sortWithinPartitions(repartitionKeyColName)
+    }
+    else {
+      repartitionedDf
+    }
+
+    optionallySortedDf.drop(repartitionKeyColName)
+
```

Comment:
```suggestion
    if (sortWithinFiles) {
      repartitionDf = repartitionedDf.sortWithinPartitions(repartitionKeyColName)
    }

    repartitionDf.drop(repartitionKeyColName)

```

Nit: keep the `var` and add the if block here. This avoids introducing a new variable.